pkg/pillar/cmd/monitor/ipc_server.go

```diff
@@ -169,14 +170,24 @@ func (s *monitorIPCServer) sendIpcMessage(t string, msg any) error {
 	defer s.Unlock()
 
 	var err error
+	var data []byte
 
-	if data, err := json.Marshal(msg); err == nil {
+	if data, err = json.Marshal(msg); err == nil {
```

Comment:
I think this pattern is more common and easier to read, but doesn't change anything logically of course.
```suggestion
	if data, err = json.Marshal(msg); err != nil {
            return err
        }
        // the normal case
```

Reply:
@europaul I thought about it again and I think my "C" legacy forces me to do so. I agree that in Go it doesn't make sense.